pkg/rest/endpoints/put_test.go

```diff
@@ -10,28 +10,37 @@ import (
 	"encoding/json"
 	"github.com/emicklei/go-restful"
 	"golang.org/x/net/context"
+	"gopkg.in/yaml.v2"
 	"io"
 	"io/ioutil"
 	"net/http/httptest"
 	"net/url"
 	"strings"
 )
 
-func marshal(payload interface{}) io.ReadCloser {
+func marshalToJSON(payload interface{}) io.ReadCloser {
 	var b []byte
 	buffer := bytes.NewBuffer(b)
 	json.NewEncoder(buffer).Encode(payload)
 	return ioutil.NopCloser(buffer)
 }
 
+func marshalToYAML(payload interface{}) io.ReadCloser {
+	var b []byte
+	buffer := bytes.NewBuffer(b)
+	raw, _ := yaml.Marshal(payload)
```

Comment:
Are you swallowing an error here?  Should this one, and the JSON one above both handle errors better?